### ISSUE TITLE
Tesseract Development TODO List

### ISSUE THREAD

**Issue by mpowelson** [ros-industrial-consortium/tesseract#66]
- [x] Continuous collision checking between two moving objects
- [x] Upgrade data structure to support graphs
- [x] Upgrade to support ROS2 #43 
    - [x] Make all core package pure cmake package
    - [x] Create ROS2 meta package containing same package in tesseract_ros meta package
- [ ] ~~Upgrade to use Bullet3~~ 
- [x] Add tesseract process planning
- [x] Update Clang format in CI from 5.0 to 8.0 #93
- [x] Unify syntax for defining shared pointer types 
- [x] Rename BasicPlanner class to Planner
- [ ] Add sub graph functionality to scene graph
- [ ] Add sub graph solver functionality to tesseract_environment. The idea behind this to mimic actual hardware in the environment.
- [x] Planner: Add reference frame to cartesian waypoint #125

**Comment by mpowelson** [comment]
I believe the ROS 2 wrapper is underway. @schornakj  could you link to it here?

**Comment by schornakj** [comment]
@mpowelson The ROS2 versions of the `tesseract_ros` packages will live here: [https://github.com/ros-industrial-consortium/tesseract_ros2](https://github.com/ros-industrial-consortium/tesseract_ros2)

The main repo is still waiting on some PRs to be merged. In the meantime, my latest dev branch is here: [https://github.com/schornakj/tesseract_ros2/tree/feature/migrate-tesseract-monitoring]( https://github.com/schornakj/tesseract_ros2/tree/feature/migrate-tesseract-monitoring)

**Comment by mpowelson** [comment]
@Levi-Armstrong Is anything here still relevant or should we close?

**Comment by Levi-Armstrong** [comment]
I would just move the sub graph items to the new one and we can mark themes nice to have. We already support adding scene graphs to another but the are not modeled as a sub graph that boost does support.

**Comment by mpowelson** [comment]
All issues have been addressed with the exception of #420.
